src/anndata/_core/merge.py

```diff
@@ -1094,11 +1114,184 @@ def concat_Xs(adatas, reindexers, axis, fill_value):
+                    else col
+                )
+                for name, col in a.items()
+            }
+        )
+
+
+DS_CONCAT_DUMMY_INDEX_NAME = "concat_index"
+
+
+def concat_dataset2d_on_annot_axis(
+    annotations: Iterable[Dataset2D],
+    join: Join_T,
+) -> Dataset2D:
+    """Create a concatenate dataset from a list of :class:`~anndata.experimental.backed._xarray.Dataset2D` objects.
+    The goal of this function is to mimic `pd.concat(..., ignore_index=True)` so has some complicated logic
+    for handling the "index" to ensure (a) nothing is loaded into memory and (b) the true index is always tracked.
+
+    Parameters
+    ----------
+    annotations
+        The :class:`~anndata.experimental.backed._xarray.Dataset2D` objects to be concatenated.
+    join
+        Type of join operation
+
+    Returns
+    -------
+    Concatenated :class:`~anndata.experimental.backed._xarray.Dataset2D`
+    """
+    from anndata._io.specs.lazy_methods import DUMMY_RANGE_INDEX_KEY
+    from anndata.experimental.backed._compat import Dataset2D
+    from anndata.experimental.backed._compat import xarray as xr
+
+    annotations_re_indexed = []
+    for a in make_xarray_extension_dtypes_dask(annotations):
+        old_key = list(a.coords.keys())[0]
+        # First create a dummy index
+        a.coords[DS_CONCAT_DUMMY_INDEX_NAME] = (
+            old_key,
+            pd.RangeIndex(a[a.attrs["indexing_key"]].shape[0]).astype("str"),
+        )
+        # Set all the dimensions to this new dummy index
+        a = a.swap_dims({old_key: DS_CONCAT_DUMMY_INDEX_NAME})
+        # Move the old coordinate into a variable
+        old_coord = a.coords[old_key]
+        del a.coords[old_key]
+        a[old_key] = old_coord
+        annotations_re_indexed.append(a)
+    # Concat along the dummy index
+    ds = Dataset2D(
+        xr.concat(annotations_re_indexed, join=join, dim=DS_CONCAT_DUMMY_INDEX_NAME),
+        attrs={"indexing_key": f"true_{DS_CONCAT_DUMMY_INDEX_NAME}"},
+    )
+    ds.coords[DS_CONCAT_DUMMY_INDEX_NAME] = pd.RangeIndex(
+        ds.coords[DS_CONCAT_DUMMY_INDEX_NAME].shape[0]
+    ).astype("str")
+    # Drop any lingering dimensions (swap doesn't delete)
+    ds = ds.drop_dims(d for d in ds.dims if d != DS_CONCAT_DUMMY_INDEX_NAME)
+    # Create a new true index and then delete the columns resulting from the concatenation for each index.
+    # This includes the dummy column (which is neither a dimension nor a true indexing column)
+    index = xr.concat(
+        [a[a.attrs["indexing_key"]] for a in annotations_re_indexed],
+        dim=DS_CONCAT_DUMMY_INDEX_NAME,
+    )
+    # prevent duplicate values
+    index.coords[DS_CONCAT_DUMMY_INDEX_NAME] = ds.coords[DS_CONCAT_DUMMY_INDEX_NAME]
+    ds[f"true_{DS_CONCAT_DUMMY_INDEX_NAME}"] = index
+    for key in set(a.attrs["indexing_key"] for a in annotations_re_indexed):
+        del ds[key]
+    if DUMMY_RANGE_INDEX_KEY in ds:
+        del ds[DUMMY_RANGE_INDEX_KEY]
+    return ds
```

Comment:
This is new since last reivew, but hopefully is clear in what it is doing especially with the docstring - basically, it creates its own index to prevent loading indices into memory unecessarily or having to worry about concatenating the indices (which matches the in-memory behavior)